### ISSUE TITLE
Static fields are included when jsonLibrary=gson 

### ISSUE THREAD

**Issue by blutorange** [vojtechhabarta/typescript-generator#496]
When `jsonLibrary=gson` is set, the type declarations for a Java interface includes its static members.

```java
public class Demo {
  public static String THIS_FIELD_SHOULD_NOT_BE_INCLUDED;
  public String thisShouldBeIncluded;
}
```

results in something like (when generating classes as interfaces)

```typescript
export interface Demo {
  THIS_FIELD_SHOULD_NOT_BE_INCLUDED: string ;
  thisShouldBeIncluded: string;
}
```

Unless I'm missing something and this is by design (?), they should not be included, at least not when generating typescript interfaces:

* The default `new Gson().toJson(someObjectImplementingTheInterfaceWithStaticField)` does not seem to includde static fields
* TypeScript does not support static fields in interfaces
* `jsonLibrary=jackson2` for static fields also does not seem to include static fields on interfaces

A quick look at the code in the debugger seems like a `Modifier.isStatic` check in the GsonParser should be enough to solve this:
 
![image](https://user-images.githubusercontent.com/2456413/85935152-3128cb00-b8e5-11ea-8d46-b55fb94fa2fa.png)

**Comment by blutorange** [comment]
For the time being, this library has some very nice hooks that lets us modify the generation process : ) That makes it possible to exclude static fields with a custo filter:

```java
package de.xima.fc.workflow.designer;

import java.lang.reflect.Modifier;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

import cz.habarta.typescript.generator.Extension;
import cz.habarta.typescript.generator.compiler.ModelCompiler.TransformationPhase;
import cz.habarta.typescript.generator.compiler.ModelTransformer;
import cz.habarta.typescript.generator.compiler.SymbolTable;
import cz.habarta.typescript.generator.emitter.EmitterExtensionFeatures;
import cz.habarta.typescript.generator.emitter.TsModel;
import cz.habarta.typescript.generator.parser.BeanModel;
import cz.habarta.typescript.generator.parser.Model;

/**
 * An {@link Extension} that excludes all static fields of interfaces and classes.
 */
public class ExcludeStaticFieldsExtension extends Extension {

  @Override
  public List<TransformerDefinition> getTransformers() {
    return Arrays.asList(new TransformerDefinition(TransformationPhase.BeforeTsModel, new ModelTransformer() {

      @Override
      public TsModel transformModel(SymbolTable symbolTable, TsModel model) {
        return model;
      }

      @Override
      public Model transformModel(SymbolTable symbolTable, Model model) {
        final List<BeanModel> newBeans = model.getBeans() //
            .stream() //
            .map(bean -> bean.withProperties( //
                bean.getProperties() //
                    .stream() //
                    .filter(prop -> !Modifier.isStatic(prop.getOriginalMember().getModifiers())) //
                    .collect(Collectors.toList()))) //
            .collect(Collectors.toList());
        return new Model(newBeans, model.getEnums(), model.getRestApplications());
      }
    }));
  }

  @Override
  public EmitterExtensionFeatures getFeatures() {
    final EmitterExtensionFeatures features = new EmitterExtensionFeatures();
    features.worksWithPackagesMappedToNamespaces = true;
    return features;
  }
}
```

**Comment by vojtechhabarta** [comment]
Gson is configurable using `excludeFieldsWithModifiers` method in `GsonBuilder`. By default it excludes fields that are `static` or `transient`. Typescript-generator should by default behave the same. I changed it but it is breaking change. For those who want to generate also `static` fields as before I added `gsonConfiguration/excludeFieldsWithModifiers` parameter. Here is Maven example:
``` xml
<configuration>
    <jsonLibrary>gson</jsonLibrary>
    <gsonConfiguration>
        <excludeFieldsWithModifiers>transient</excludeFieldsWithModifiers>  <!-- static not excluded -->
    </gsonConfiguration>
    ...
</configuration>
```
@blutorange thanks for this issue and sharing your workaround.
